jquery.flot.legend.js

```diff
@@ -34,21 +34,24 @@
             return;
         }
 
-        var fragments = [],
-            entries = [],
-            rowStarted = false,
+        var entries = [],
             lf = options.legend.labelFormatter,
-            s, label;
+            s, label, i;
 
         // Build a list of legend entries, with each having a label and a color
-        for (var i = 0; i < series.length; ++i) {
```

Comment:
I recommend refactoring this for statement using array **filter** and **map** functions.